core/local/local.go

```diff
@@ -337,7 +339,10 @@ func (e *ExecutionScheduler) runExecutor(
 
 // Run the ExecutionScheduler, funneling all generated metric samples through the supplied
 // out channel.
-func (e *ExecutionScheduler) Run(globalCtx, runCtx context.Context, engineOut chan<- stats.SampleContainer) error {
+//nolint:cyclop
+func (e *ExecutionScheduler) Run(
+	globalCtx, runCtx context.Context, engineOut chan<- stats.SampleContainer, builtinMetrics *metrics.BuiltinMetrics,
```

Comment:
Why do we pass the `BuiltinMetrics` here, instead of in the `ExecutionScheduler` constructor? :confused: Wouldn't it be clearer to pass it to the `NewExecutor()` constructor, e.g. through the `ExecutionState`? https://github.com/grafana/k6/blob/0628db0553c1938a52fcde01a83326207748f32f/lib/executors.go#L106